### PR TITLE
Move unversioned.GroupVersion to pkg/runtime/schema

### PR DESCRIPTION
**Pull request by smarterclayton**
@caesarxuchao extracted from #36121, will my refactor PR after yours

@thockin as discussed about breaking circular dependencies.  More will follow

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/36782)

### CODE REVIEW

File: convert.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+cat ${files_to_convert} | xargs -n1 sed -i "s|unversioned\\.GroupKind|schema\\.GroupKind|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|unversioned\\.ObjectKind|schema\\.ObjectKind|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|unversioned\\.FromAPIVersionAndKind|schema\\.FromAPIVersionAndKind|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|unversioned\\.EmptyObjectKind|schema\\.EmptyObjectKind|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|unversioned\\.ParseGroupVersion|schema\\.ParseGroupVersion|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|unversioned\\.ParseResourceArg|schema\\.ParseResourceArg|g"
+echo pkg/client/typed/discovery/helper_blackbox_test.go | xargs -n1 sed -i "s|uapi\\.GroupVersion|schema\\.GroupVersion|g"
+cat ${files_to_convert} | xargs -n1 sed -i "s|schema\\.GroupVersionForDiscovery|unversioned\\.GroupVersionForDiscovery|g"
+goimports -w ./pkg ./federation ./test ./plugin ./cmd
+echo test/e2e_node/apparmor_test.go | xargs -n1 sed -i "s|k8s\\.io/kubernetes/pkg/runtime/schema|k8s\\.io/client-go/pkg/runtime/schema|g"
```
Comment (caesarxuchao) [inline convert.sh]: Why client-go?

Comment (smarterclayton) [inline convert.sh]: Errors package in that test comes from client-go, you must pass runtime/schema from client package to that method.

Comment (caesarxuchao) [inline convert.sh]: I confirmed with @timstclair client-go was imported by accident. Suggest import the kubernetes packages.

**Comment by thockin** [comment]
I tagged you on the PR for a simple way for me to break the cycle, too.
Much smaller than this PR :)

2016-11-14 15:25 GMT-08:00 Clayton Coleman <notifications@github.com>:

**Comment by caesarxuchao** [comment]
@smarterclayton the changes look good. How much more do you have locally?

**Comment by smarterclayton** [comment]
I rebased on top of your branch and dropped convert.sh out as well as fixing a few small things that the internal -> versioned change caused.  I'll also drop the "staging/copy.sh on Mac" commit since it's possible that breaks Linux and it was just local.

**Comment by caesarxuchao** [comment]
e2e:
```
W1123 13:31:47.898] test/e2e/network_partition.go:371: cannot use StatefulSetGroupVersionResource (type "k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type "k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:83: cannot use StatefulSetGroupVersionResource (type "k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type "k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:411: cannot use StatefulSetGroupVersionResource (type "k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type "k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in argument to framework.SkipIfMissingResource
```

**Comment by smarterclayton** [comment]
Yeah, I can't compile on mac right now because of the cross compile issue
so I suspect my script didn't catch it.

On Nov 23, 2016, at 3:38 PM, Chao Xu <notifications@github.com> wrote:

e2e:

W1123 13:31:47.898] test/e2e/network_partition.go:371: cannot use
StatefulSetGroupVersionResource (type
"k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type
"k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in
argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:83: cannot use
StatefulSetGroupVersionResource (type
"k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type
"k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in
argument to framework.SkipIfMissingResource
W1123 13:31:47.899] test/e2e/petset.go:411: cannot use
StatefulSetGroupVersionResource (type
"k8s.io/kubernetes/pkg/api/unversioned".GroupVersionResource) as type
"k8s.io/kubernetes/pkg/runtime/schema".GroupVersionResource in
argument to framework.SkipIfMissingResource

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/36782#issuecomment-262636230>,
or mute the thread

**Comment by smarterclayton** [comment]
Actually this is because your branch isn't up to date with latest.

**Comment by saad-ali** [comment]
Ok I just merged Chao's PR https://github.com/kubernetes/kubernetes/pull/36673. Clayton, feel free to rebase. Ping me when you're done. I will be back online around 8:30 PM PST and will merge your PR then, if the queue is still green.

**Comment by saad-ali** [comment]
@k8s-bot verify test this

**Comment by smarterclayton** [comment]
There were actual verification failures, pushing another change now.

**Comment by smarterclayton** [comment]
All green

**Comment by saad-ali** [comment]
Merging

**Pull request merged**